### ISSUE TITLE
simplified git config causes remote upstream rewrite on every run

### ISSUE THREAD

**Issue by aschleifer** [vcs-python/vcspull#332]
If I use a simple config entry like:

```
~/test/cvspull:
  mypackage: 'git+ssh://git@gitlab.com/mygroup/mypackage.git'
```

I always see the line about `Updating remote origin`:
```
Updating remote origin (ssh://git@gitlab.com/mygroup/mypackage.git) with git+ssh://git@gitlab.com/mygroup/mypackage.git
```
which then in turn causes a retry of the update.

To fix this issue I have to use a config like this:
```
~/test/cvspull:
  mypackage:
    url: 'git+ssh://git@gitlab.com/mygroup/mypackage.git'
    remotes:
      origin: 'ssh://git@gitlab.com/mygroup/mypackage.git'
```

This then works, but I would like to have a more compact/simplified config with only one line per repo.

(repogroup and name have been abstracted, as the used gitlab is internal ;) )

Best regards
Segaja

**Comment by tony** [comment]
@aschleifer Thank you, good catch!

If you want to start a pull request you are welcome.  If not I will hopefully get around to it soon - I've also been getting it

**Comment by aschleifer** [comment]
I don't think I'm deep enough in the code to easily do it right now and my python is a bit rusty.

**Comment by tony** [comment]
@aschleifer No problem, it's about time I give a tests a top-down review

**Comment by aschleifer** [comment]
@tony once this bug is fixed I'm happy to fix the new scripts to use the simplified configuration again. But you can also do it in the same PR where you fix this.

**Comment by aschleifer** [comment]
@tony just a headsup: since now the new structure is in, i will wait for this fix before i look at the `add` command and into moving the gitlab generator script into the actual package.

**Comment by tony** [comment]
@aschleifer Noted, and looking into this during the weekend

**Comment by tony** [comment]
@aschleifer Fix is still a WIP

Will continue next weekend

After this issue and #338 is fixed, will move onto making libvcs and vcspull easier to read.  I originally started this 8 years ago when totally new at python.

**Comment by tony** [comment]
@aschleifer If you try `pip install --user -U vcspull`, is the prerelease any better?

Fix is live is in 1.12.0b0

**Comment by tony** [comment]
@aschleifer Better with 1.12.0?
